src/plat/imx6/config.cmake

```diff
@@ -8,26 +9,31 @@ cmake_minimum_required(VERSION 3.7.2)
 
 declare_platform(imx6 KernelPlatImx6 PLAT_IMX6 KernelSel4ArchAarch32)
 
-set(c_configs PLAT_SABRE PLAT_WANDQ)
-set(cmake_configs KernelPlatformSabre KernelPlatformWandQ)
-set(plat_lists sabre wandq)
-foreach(config IN LISTS cmake_configs)
-    unset(${config} CACHE)
```

Comment:
This loop was here in order to be able to switch KernelArmPlatfrom in the build config editor without the previous config option remaining in the cache.  For example if you first select sabre, and then later re-select to wandq, the KernelPlatformSabre and KernelPlatImx6dq options need to be unset otherwise they will remain in the cache and can cause false positive platform checks to occur.

Reply:
Good to know. That is a workflow we are not using, we always have distinct workspaces (build directories) for each configuration. Plus the workspace is always wiped when a change in the build parameters is detected. We found that this is the the best way, as switching on the same workspace yields surprises sometimes. 
If you want to keep this switching feature, I have to rework some things also in other PR. But I have another branch that changes `declare_platform()` and I might be able to keep this working then. Haven made it a PR yet,see https://github.com/axel-h/seL4/commits/patch-axel-13

Reply:
I would like to keep this behavior.  ( I did find myself thinking the other day that maybe the kernel build could get a large refactor to not use the cache at all which would avoid these state effects).

Reply:
@kent-mcleod I've change it that the variables are cleared